HW3

watchbot23 opened the pull request.

Commit:
Added HW3

LiDmYr commented on GDM.HW3.ConditionalsMethods/Program.cs:
```diff
@@ -0,0 +1,315 @@
+            do
+            {
+                while (array[f] < mid)
+                    ++f;
```
Please use brackets with _while_, _for_, _if_, etc.

LiDmYr commented on GDM.HW3.ConditionalsMethods/Program.cs:
```diff
@@ -0,0 +1,315 @@
+            }
+            return temp;
+        }
+        static void Main(string[] args)
```
Main is a main method - it should be the first among all methods

LiDmYr commented on GDM.HW3.ConditionalsMethods/Program.cs:
```diff
@@ -0,0 +1,315 @@
+            while (eggs < eggsNumberInBox)
+            {
+                eggs++;
+
```
What is an aim of this empty line ?

LiDmYr commented on GDM.HW3.ConditionalsMethods/Program.cs:
```diff
@@ -0,0 +1,315 @@
+            {
+                eggs--;
+                count++;
+
```
What is an aim of this empty line ?

LiDmYr commented on GDM.HW3.ConditionalsMethods/Program.cs:
```diff
@@ -0,0 +1,315 @@
+            Console.WriteLine(task3 + $"Последовательностью чисел Фибоначчи для {fib} является:");
+            for (int i = 0; i < fib; i++)
+            {
+                Console.Write(Fibonachi(i) + " ");
```
You repeated the same calculation a lot of times. Try to call only one time this function.

LiDmYr commented on GDM.HW3.ConditionalsMethods/Program.cs:
```diff
@@ -0,0 +1,315 @@
+        //Task 2
+        static double Factorial(double fac)
+        {
+            if (fac == 0)
```
use ternary operator + handle a case when 'fac = -1', etc

LiDmYr commented on GDM.HW3.ConditionalsMethods/Program.cs:
```diff
@@ -0,0 +1,315 @@
+        //Task 4
+        static int DaysNumberInMonth(int month, int year)
+        {
+            if (month == 2 && year % 4 == 0 && year % 100 != 0 || month == 2 && year % 100 == 0)
```
Use if(...){...}else if(...){...}etc. - of course you use 'return' and it's not possible to go to another if(...) but it looks "strange"

LiDmYr commented on GDM.HW3.ConditionalsMethods/Program.cs:
```diff
@@ -0,0 +1,315 @@
+            }
+            else
+            {
+                return 31;
```
Try to assign variable and return one time

LiDmYr commented on GDM.HW3.ConditionalsMethods/Program.cs:
```diff
@@ -0,0 +1,315 @@
+        }
+
+        //Task 5
+        static int[] ReccursiveQuickArraySort(int[] array, int firstArrayIndex, int lastArrayIndex)
```
Your algorithm is faster in 5-10 times than my example in github :) GREAT JOB:)

watchbot23 replied:
wow)

LiDmYr commented on GDM.HW3.ConditionalsMethods/Program.cs:
```diff
@@ -0,0 +1,315 @@
+        //Task 5
+        static int[] ReccursiveQuickArraySort(int[] array, int firstArrayIndex, int lastArrayIndex)
+        {
+            int f = firstArrayIndex;
```
Try to use human readable variables names

LiDmYr commented on GDM.HW3.ConditionalsMethods/Program.cs:
```diff
@@ -0,0 +1,315 @@
+            int min = 0;
+            int max = 0;
+
+            for (int i = 0; i < array.Length; i++)
```
Why do use 2 time the same for loop on 95 + 102 lines?

watchbot23 replied:
cos I don't know if the max value has bigger index than min one or not.

LiDmYr commented on GDM.HW3.ConditionalsMethods/Program.cs:
```diff
@@ -0,0 +1,315 @@
+                    max = i;
+                }
+            }
+            if (min > max)
```
Rewrite 109-122 lines with only one _if(){}else{}_ and only one _for_

LiDmYr commented on GDM.HW3.ConditionalsMethods/Program.cs:
```diff
@@ -0,0 +1,315 @@
+                if (array[i] == veriable && arrayIndex != i)
+                {
+                    temp = true;
+                    return temp;
```
You don't need to **return temp** (you already have this on 138 line) here - just break for.

watchbot23 replied:
oops. I know. Just fogot to delete it.

LiDmYr commented on GDM.HW3.ConditionalsMethods/Program.cs:
```diff
@@ -0,0 +1,315 @@
+        //Task 8
+        static bool IsDupl(int[] array, int veriable, int arrayIndex)
+        {
+            bool temp = false;
```
Write another name - **_isSomethingHappened_**

LiDmYr reviewed (commented):
I would like you implement my notes/comments

watchbot23 committed:
deleted for() from         static int[] PutZeroBetweenMimMaxValueArray() method

watchbot23 committed:
Added new method FibonachiArray

watchbot23 committed:
Factorial() changed with using ternar conditions.
Added the a case when 'fac = -1', etc

watchbot23 committed:
deleted ...else{} from DaysNumberInMonth()

watchbot23 committed:
- deleted some extra raws without any influence on code
- renamed some veriables with humanreadable values
- renamed IsDupl() to IsArrayVeriableDuplicated()

watchbot23 committed:
Added 2 more variants for Fibonacci()

Commit:
No changes

Commit:
Created new method ConsoleWriteArraysArray()
created new ConsoleApp project RadMap

Commit:
Merge branch 'HW3' of https://github.com/watchbot23/GDM.UIP.HW into HW3

LiDmYr commented on GDM.HW3.ConditionalsMethods/Program.cs:
```diff
@@ -0,0 +1,341 @@
+        //Task 8
+        static bool IsArrayVeriableDuplicated(int[] array, int veriable, int arrayIndex)
+        {
+            bool temp = false;
```
please change the name

LiDmYr commented on GDM.HW3.ConditionalsMethods/Program.cs:
```diff
@@ -0,0 +1,341 @@
+                if (array[i] == veriable && arrayIndex != i)
+                {
+                    temp = true;
+                }
```
why do you need to continue for loop if you already have the 'true' ?

LiDmYr reviewed (commented):
Actually not all changes which I asked you made
